pyClient/test_commands/scenario.py

```diff
@@ -153,6 +155,45 @@ def charlie_double_withdraw(
     note1_value = joinsplit.to_zeth_units(EtherValue(CHARLIE_WITHDRAW_CHANGE_ETH))
     v_out = EtherValue(CHARLIE_WITHDRAW_ETH)
 
+    # ### ATTACK BLOCK
+    # Add malicious nullifiers: we reuse old nullifiers to double spend by
+    # adding $r$ to them so that they have the same value as before in Z_r,
+    # and so the zksnark verification passes, but have different values in
+    # {0;1}^256 so that they appear different to the contract.
+    # See: https://github.com/clearmatics/zeth/issues/38
+
+    def compute_h_sig_attach_nf(
```

Comment:
what does `_attach_nf` mean?

Reply:
change to `compute_h_sig_attack_nf` please

Reply:
SHould have been `attack`.  Renamed now.